resources/assets/js/components/ApplicantProfile/Skills/List.tsx

```diff
@@ -0,0 +1,202 @@
+    description: "Text for the 'Journey' option on the sort toggle UI.",
+  },
+  sortBy: {
+    id: "applicantProfile.skills.list.sortBy",
+    defaultMessage: "Sort By:",
+    description: "Text for the label on the sort toggle UI.",
+  },
+});
+
+enum SortTypes {
+  Group = "group",
+  Alpha = "alpha",
+  // Journey = "journey", Future enhancement according to designs.
+}
+interface SortButtonProps {
+  text: string;
+  active: boolean;
+  handleClick: () => void;
+}
+
+const SortButton: React.FC<SortButtonProps> = ({
+  text,
+  active,
+  handleClick,
+}) => (
+  <button
+    data-h2-button={`${active ? "theme-1" : "black"}, square, medium, clear`}
+    type="button"
+    onClick={handleClick}
+  >
+    <span data-h2-button-label>{text}</span>
+  </button>
+);
+interface ListProps {
+  experiences: Experience[];
+  experienceSkills: ExperienceSkill[];
+  skillCategories: SkillCategory[];
+  skillSkillCategories: SkillSkillCategory[];
+  skills: Skill[];
+  applicantId: number;
+}
+
+const List: React.FC<ListProps> = ({
+  experiences,
+  experienceSkills,
+  skillCategories,
+  skillSkillCategories,
+  skills,
+  applicantId,
+}) => {
+  const intl = useIntl();
+  const locale = getLocale(intl.locale);
+  const [sortType, setSortType] = useState<SortTypes>(SortTypes.Group);
+
+  const skillCategoriesGroupedObject = skillCategories
+    .filter((skillCategory) => skillCategory.depth !== 1) // Filter out non-top-level categories.
+    .reduce(
+      (
+        accumulator: {
+          [catId: number]: {
+            title: localizedFieldNonNull | undefined;
+            skills: Skill[];
+          };
+        },
+        currentCategory,
+      ) => {
+        if (currentCategory.parent_id !== null) {
+          // Ensure there is a parent category set.
+          const titleValue = skillCategories
+            .filter((skillCategory) => skillCategory.depth === 1)
+            .find((x) => x.id === currentCategory.parent_id)?.name;
+
+          const skillsValue = skills.filter(
+            (currentSkill) =>
+              currentSkill.id ===
+              skillSkillCategories.find(
+                (y) =>
+                  y.skill_id === currentSkill.id &&
+                  y.skill_category_id === currentCategory.id,
+              )?.skill_id,
+          );
+
+          if (hasKey(accumulator, currentCategory.parent_id)) {
+            accumulator[currentCategory.parent_id].skills = [
+              ...skillsValue,
+              ...accumulator[currentCategory.parent_id].skills,
+            ]; // Add child category to existing parent category group.
+          } else {
+            accumulator[currentCategory.parent_id] = {
+              title: titleValue,
+              skills: skillsValue,
+            }; // Add child category to new parent category group.
+          }
+        }
+        return accumulator;
+      },
+      [],
+    );
+
+  const skillCategoriesGrouped = Object.keys(skillCategoriesGroupedObject).map(
```

Comment:
I think you want filter instead of map here. Having entries of `false` in the resulting array isn't useful.